src/solvers/flattening/bv_pointers.cpp

```diff
@@ -254,11 +254,10 @@ bvt bv_pointerst::convert_pointer_type(const exprt &expr)
     if(op_type.id()==ID_pointer)
       return convert_bv(op);
     else if(
-      op_type.id() == ID_signedbv || op_type.id() == ID_unsignedbv ||
-      op_type.id() == ID_bool || op_type.id() == ID_c_enum ||
-      op_type.id() == ID_c_enum_tag || op_type.id() == ID_bv)
+      can_cast_type<bitvector_typet>(op_type) || op_type.id() == ID_c_enum ||
+      op_type.id() == ID_c_enum_tag)
     {
```

Comment:
Note that ID_bool is not a bit vector type, i.e., this case is lost.

Reply:
Ah, thanks, fixed!